site/app/views/forum/ForumThreadView.php

```diff
@@ -1121,7 +1121,8 @@ public function createPost($thread_id, $post, $unviewed_posts, $first, $reply_le
             "parent_id" => $parent_id,
             "render_markdown" => $markdown,
             "has_history" => $has_history,
-            "thread_previously_merged" => $merged_thread
+            "thread_previously_merged" => $merged_thread,
+            "thread_announced" => $this->core->getQueries()->existsAnnouncementsId($thread_id)
```

Comment:
This query should happen within the controller and then pass in the value to the view.